exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +73,2905 @@ func TestStart(t *testing.T) {
+	_, err = g.FetchTradablePairs(context.Background(), asset.CrossMargin)
+	if err != nil {
+		t.Errorf("%s FetchTradablePairs() error %v", g.Name, err)
+	}
+}
+
+func TestUpdateTradablePairs(t *testing.T) {
+	t.Parallel()
+	if err := g.UpdateTradablePairs(context.Background(), true); err != nil {
+		t.Errorf("%s UpdateTradablePairs() error %v", g.Name, err)
+	}
+}
+
+func TestUpdateTickers(t *testing.T) {
+	t.Parallel()
+	if err := g.UpdateTickers(context.Background(), asset.DeliveryFutures); err != nil {
+		t.Errorf("%s UpdateTickers() error %v", g.Name, err)
+	}
+	if err := g.UpdateTickers(context.Background(), asset.Futures); err != nil {
+		t.Errorf("%s UpdateTickers() error %v", g.Name, err)
+	}
+	if err := g.UpdateTickers(context.Background(), asset.Spot); err != nil {
+		t.Errorf("%s UpdateTickers() error %v", g.Name, err)
+	}
+	if err := g.UpdateTickers(context.Background(), asset.Options); err != nil {
+		t.Errorf("%s UpdateTickers() error %v", g.Name, err)
+	}
+	if err := g.UpdateTickers(context.Background(), asset.CrossMargin); err != nil {
+		t.Errorf("%s UpdateTickers() error %v", g.Name, err)
+	}
+	if err := g.UpdateTickers(context.Background(), asset.Margin); err != nil {
+		t.Errorf("%s UpdateTickers() error %v", g.Name, err)
+	}
+}
+
+func TestUpdateOrderbook(t *testing.T) {
+	t.Parallel()
+	cp, err := getFirstTradablePair(t, asset.Futures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := g.UpdateOrderbook(context.Background(), cp, asset.Futures); err != nil {
+		t.Errorf("%s UpdateOrderbook() error %v", g.Name, err)
+	}
+}
+
+func TestGetWithdrawalsHistory(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.GetWithdrawalsHistory(context.Background(), currency.BTC, asset.Empty); err != nil {
+		t.Errorf("%s GetWithdrawalsHistory() error %v", g.Name, err)
+	}
+}
+func TestGetRecentTrades(t *testing.T) {
+	t.Parallel()
+	currencyPair := currency.NewPair(currency.BTC, currency.USDT)
+	_, err := g.GetRecentTrades(context.Background(), currencyPair, asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+	cp, err := getFirstTradablePair(t, asset.DeliveryFutures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), cp, asset.DeliveryFutures)
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), currencyPair, asset.Options)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSubmitOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	enabledPairs, err := g.GetEnabledPairs(asset.CrossMargin)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var orderSubmission = &order.Submit{
+		Exchange:  g.Name,
+		Pair:      enabledPairs[0],
+		Side:      order.Buy,
+		Type:      order.Limit,
+		Price:     1,
+		Amount:    1,
+		AssetType: asset.CrossMargin,
+	}
+	_, err = g.SubmitOrder(context.Background(), orderSubmission)
```

Comment:
fee currency is not being set. [previous nit](https://github.com/thrasher-corp/gocryptotrader/pull/1058#discussion_r1051687820)